File: test/fixtures/javascript/class-methods.js
class UserService {
  constructor(db) {
    this.db = db;
  }

  async fetchUser(id) {
    const user = await this.db.findById(id);
    if (!user) throw new Error('User not found');
    return user;
  }

  async updateUser(id, data) {
    const user = await this.fetchUser(id);
    Object.assign(user, data);
    await this.db.save(user);
    return user;
  }

  async deleteUser(id) {
    const user = await this.fetchUser(id);
    await this.db.remove(user);
    return { deleted: true };
  }

  validateEmail(email) {
    const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    return regex.test(email);
  }
}

const helpers = {
  formatName: (first, last) => `${first} ${last}`,
  capitalize: (str) => str.charAt(0).toUpperCase() + str.slice(1),
};

File: test/fixtures/python/class-methods.py
class UserService:
    def __init__(self, db):
        self.db = db

    def fetch_user(self, user_id: str):
        user = self.db.find_by_id(user_id)
        if not user:
            raise ValueError("User not found")
        return user

    def update_user(self, user_id: str, data: dict):
        user = self.fetch_user(user_id)
        for key, value in data.items():
            setattr(user, key, value)
        self.db.save(user)
        return user

    def delete_user(self, user_id: str):
        user = self.fetch_user(user_id)
        self.db.remove(user)
        return {"deleted": True}

    def validate_email(self, email: str) -> bool:
        import re
        pattern = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
        return bool(re.match(pattern, email))


def top_level_function():
    return "I am at the top level"


def another_top_level():
    return "Another top level function"
